### PR TITLE
Change SFirstOfNextMonth to SFirstOfMonth w/ offset

### PR DESCRIPTION
**Pull request by deetergp**
cc @coleaeason @tylerkaraszewski

### CODE REVIEW

File: libstuff/STime.cpp [hunk 1]
```diff
@@ -55,11 +55,15 @@ string SCURRENT_TIMESTAMP_MS() {
     return timestamp + "." + msString;
 }
 
-string SFirstOfNextMonth(const string& timeStamp) {
+string SFirstOfMonth(const string& timeStamp, const int64_t& offset) {
+
+    if (offset < 0) {
+        STHROW("500 Offset may only be positive");
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:61]: Why? Let's make this general purpose and let it work for any month, including last month or whenever.

In fact, let's call it `SFirstOfMonth` and pass `timestamp` and `monthsInFuture`. If you pass `0`, you get the first of this month, if you pass `-1` you get last month. If you pass `1` you get next month, if you pass `12` you get this month next year, etc. Set the default for `monthsInFuture` to 0 or 1, your choice.

File: libstuff/STime.cpp [hunk 2]
```diff
@@ -81,11 +85,11 @@ string SFirstOfNextMonth(const string& timeStamp) {
         // If the month is 11, that means its december, we need to roll the year
         // up by one and set the month to january. Otherwise just move the month
         // forward one.
-        if (month == 11) {
+        if ((month + offset) >= 11) {
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:88]: Make this work forward and backward and in increments of more than a year.

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]

File: libstuff/STime.cpp (hunk 1 above)
Comment (coleaeason) [reply libstuff/STime.cpp:61]: Agree with tyler -- we should just make this a general date manipulation function for finding the first of any month in relation to the current timestamp

File: libstuff/STime.cpp [hunk 3]
```diff
@@ -81,11 +85,11 @@ string SFirstOfNextMonth(const string& timeStamp) {
...
             t.tm_year += 1;
-            t.tm_mon = 0;
+            t.tm_mon = (month + offset) - 12;
         } else {
-            t.tm_mon = month + 1;
+            t.tm_mon = month + offset;
```
Comment (coleaeason) [inline libstuff/STime.cpp:92]: you repeat `month + offset` like 3 times, might as well make it a variable

**Comment by deetergp** [comment]
Updated! Cleaned up with a ton of help from @tyler.

**Review by tylerkaraszewski — APPROVED** [review]

**Review by coleaeason — APPROVED** [review]

**Pull request merged**